synapse/config/_util.py

```diff
@@ -64,3 +65,30 @@ def json_error_to_config_error(
         else:
             path.append(str(p))
     return ConfigError(e.message, path)
+
+
+Model = TypeVar("Model", bound=BaseModel)
+
+
+def validate_instance_map_config(
```

Comment:
> It's possible(although unlikely) this could be used for stream_writers in the future. I know of no other 'Map' style configuration done in the WorkersConfig section where this would be appropriate.

The [`databases`](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#databases) config option springs to mind here.

Reply:
OMG, I can't believe I didn't consider anything outside of `WorkersConfig`. I think might end up moving all the potential `Model`s out into another file before this is all said and done(obviously, not this PR though).

How about `parse_and_validate_mapping()`?

Reply:
Done in [cfe048e](https://github.com/matrix-org/synapse/pull/15431/commits/cfe048ecd677bcfb04edc5cb7020c8f9e9e9af57)